tools/test

```diff
@@ -145,19 +145,15 @@ run_native_android() {
     files_check android/ \
         || return 0
 
-    (
-        cd android
-
-        ./gradlew -q :app:assembleDebug :app:assembleDebugUnitTest \
-            :app:bundleDebug || exit
-
-        # The `-q` suppresses noise from warnings about obsolete build config
-        # in our dependencies from the React Native ecosystem.
-        # But it also suppresses the names of tests that failed.
-        # So on failure, rerun without it.
-        ./gradlew -q :app:testDebugUnitTest \
-            || ./gradlew :app:testDebugUnitTest || exit
-    )
+    tools/gradle -q :app:assembleDebug :app:assembleDebugUnitTest \
+        :app:bundleDebug || exit
+
+    # The `-q` suppresses noise from warnings about obsolete build config
+    # in our dependencies from the React Native ecosystem.
+    # But it also suppresses the names of tests that failed.
+    # So on failure, rerun without it.
+    tools/gradle -q :app:testDebugUnitTest \
+        || tools/gradle :app:testDebugUnitTest || exit
```

Comment:
Ah good catch, thanks. I even did this part:

> If I introduce an Android error in `FcmMessageTest.kt`,

though not the others, and apparently didn't look close enough to notice that the other suites didn't run.

I think the fix is just that those `|| exit` should become `|| return`. (And only the one that's not the last command in the function is still needed. Likely only that one was needed in the first place.) They were `exit` because the `( … )` meant a subshell, i.e. a new shell process as a child process of the main one; but now that they're running in the main shell process (because there's no longer that `cd` that needs a subshell to contain its effects), that's the wrong thing.